skimage/transform/_geometric.py

```diff
@@ -641,37 +686,25 @@ def estimate(self, src, dst):
             True, if model estimation succeeds.
 
         """
+        n, d = src.shape
 
-        try:
-            src_matrix, src = _center_and_normalize_points(src)
-            dst_matrix, dst = _center_and_normalize_points(dst)
-        except ZeroDivisionError:
-            self.params = np.nan * np.empty((3, 3))
+        src_matrix, src = _center_and_normalize_points(src)
+        dst_matrix, dst = _center_and_normalize_points(dst)
+        if not np.all(np.isfinite(src_matrix + dst_matrix)):
+            self.params = np.full((d, d), np.nan)
             return False
 
-        xs = src[:, 0]
-        ys = src[:, 1]
-        xd = dst[:, 0]
-        yd = dst[:, 1]
-        rows = src.shape[0]
-
         # params: a0, a1, a2, b0, b1, b2, c0, c1
-        A = np.zeros((rows * 2, 9))
-        A[:rows, 0] = xs
-        A[:rows, 1] = ys
-        A[:rows, 2] = 1
-        A[:rows, 6] = - xd * xs
-        A[:rows, 7] = - xd * ys
-        A[rows:, 3] = xs
-        A[rows:, 4] = ys
-        A[rows:, 5] = 1
-        A[rows:, 6] = - yd * xs
-        A[rows:, 7] = - yd * ys
-        A[:rows, 8] = xd
-        A[rows:, 8] = yd
+        A = np.zeros((n * d, (d+1) ** 2))
+        for ddim in range(d):
```

Comment:
This may now be worth a comment on its origin.  The first version was more obvious where it came from.

Reply:
@stefanv done, see comment above this line.